### PR TITLE
Adding Github Actions

### PR DESCRIPTION
**Pull request by mikeringrose**
Enables Github Actions. Two workflow have been added:

1. `build-test` - runs on pull requests and pushes to master
2. `gh-pages` - only runs on master, after the build-test workflow completes

For the `gh-pages` workflow to run correctly, a secret needs to be created named `HUBJS_TOKEN` with the scope of repo:public.

### CODE REVIEW

**Review by drspacemanphd — APPROVED** [review]

**Comment by tomwayson** [comment]
This is awesome @mikeringrose. Can we see the output of the `build-test` on this PR?

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    
+    strategy:
+      matrix:
+        node-version: [10.x]
```
Comment (tomwayson) [inline .github/workflows/build-test.yml:17]: can we specify something like `node, lts/*` to test on the current stable, and latest LTS?

Comment (tomwayson) [reply .github/workflows/build-test.yml:17]: and should we #443

**Comment by mikeringrose** [comment]
> This is awesome @mikeringrose. Can we configure it so that we see the [output of the `build-test` action](https://github.com/Esri/hub.js/actions) inline in this PR?

The codecov output in the comments from the github action.

**Review by tomwayson — APPROVED** [review]
Now I see the link to the action output under the checks section.

I whole heartedly approve this PR. Great work @mikeringrose!

**Pull request merged**